app/models/katello/upstream_pool.rb

```diff
@@ -0,0 +1,45 @@
+require 'katello/resources/candlepin'
+
+module Katello
+  class UpstreamPool < OpenStruct
+    CP_POOL = Resources::Candlepin::UpstreamPool
+
+    class << self
+      def fetch_pools(organization, params)
+        CP_POOL.organization = organization
+        pools = JSON.parse(CP_POOL.get(params: cp_request_params.deep_merge(params)))
+        pools.map { |pool| self.new(map_attributes(pool)) }
+      ensure
+        CP_POOL.organization = nil
+      end
+
+      def map_attributes(pool)
+        {
+          pool_id: pool['id'],
+          active: pool['activeSubscription'],
+          quantity: pool['quantity'],
+          start_date: pool['startDate'],
+          end_date: pool['endDate'],
+          contract_number: pool['contractNumber'],
+          consumed: pool['consumed'],
+          product_name: pool['productName'],
+          product_id: pool['productId'],
+          subscription_id: pool['subscriptionId']
+        }
+      end
+
+      def cp_request_params
+        { include: ['id',
```

Comment:
@waldenraines This API is a bit weird as it contains no information from our database (just what upstream candlepin gives us). If we add user selectable and cacheable columns, it will be in a future feature request; however, Candlepin allows include/exclude, and while we're not exposing that to our API users, we are using it ourselves right here.